rest/admin_api_test.go

```diff
@@ -2966,3 +2976,142 @@ func TestObtainUserChannelsForDeletedRoleCasFail(t *testing.T) {
 
 	}
 }
+
+// Test warnings being issued when a new channel is created with over 250 characters - CBG-1475
+func TestChannelNameSizeWarningBoundaries(t *testing.T) {
+	syncFn := "function sync(doc, oldDoc) { channel(doc.chan); }"
+	var rt *RestTester
+
+	testCases := []struct {
+		name                string
+		warnThresholdLength uint32
+		channelLength       int
+		expectWarn          bool
+	}{
+		{
+			name:                "Over max default channel length",
+			warnThresholdLength: base.DefaultWarnThresholdChannelNameSize,
+			channelLength:       int(base.DefaultWarnThresholdChannelNameSize) + 1,
+			expectWarn:          true,
+		},
+		{
+			name:                "Equal to max default channel length",
+			warnThresholdLength: base.DefaultWarnThresholdChannelNameSize,
+			channelLength:       int(base.DefaultWarnThresholdChannelNameSize),
+			expectWarn:          false,
+		},
+		{
+			name:                "Under max default channel length",
+			warnThresholdLength: base.DefaultWarnThresholdChannelNameSize,
+			channelLength:       int(base.DefaultWarnThresholdChannelNameSize) - 1,
+			expectWarn:          false,
+		},
+		{
+			name:                "Over max configured channel length",
+			warnThresholdLength: 500,
+			channelLength:       501,
+			expectWarn:          true,
+		},
+		{
+			name:                "Equal to max configured channel length",
+			warnThresholdLength: 500,
+			channelLength:       500,
+			expectWarn:          false,
+		},
+		{
+			name:                "Under max configured channel length",
+			warnThresholdLength: 500,
+			channelLength:       499,
+			expectWarn:          false,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			var thresholdConfig db.WarningThresholds
+			// If threshold is not default then configure it
+			if test.warnThresholdLength != base.DefaultWarnThresholdChannelNameSize {
+				thresholdConfig = db.WarningThresholds{ChannelNameSize: &test.warnThresholdLength}
+			}
+			rt = NewRestTester(t, &RestTesterConfig{
+				SyncFn: syncFn,
+				DatabaseConfig: &DbConfig{
+					Unsupported: db.UnsupportedOptions{
+						WarningThresholds: thresholdConfig,
+					},
+				},
+			})
+			defer rt.Close()
+
+			chanNameWarnCountBefore := rt.ServerContext().Database("db").DbStats.Database().WarnChannelNameSizeCount.Value()
+			tr := rt.SendAdminRequest("PUT", fmt.Sprintf("/db/doc%v", test.channelLength), fmt.Sprintf("{\"chan\":\"%s\"}", strings.Repeat("A", test.channelLength)))
```

Comment:
Lots going on everywhere we're sending a request - which is difficult to glance over quickly.

I'd pull out each bit of the request into variables (which can be useful for sticking into assertion messages in some more complex test cases, good habit to get into).
You can also use backticks helps with the JSON by avoiding the escaped quotes.

```suggestion
			docID := fmt.Sprintf("/db/doc%v", test.channelLength)
			chanName := strings.Repeat("A", test.channelLength)
			tr := rt.SendAdminRequest("PUT", docID, `{"chan":"`+chanName`"}`)
```